### DISCUSSION TITLE
what is neighboring MinIO instance meaning?

### DISCUSSION THREAD

**Discussion by xiaopeifeng** [minio/minio#17159] [Q&A] [answered]
I'm reading a document in the historical version and I don't quite understand exactly what is meant by neighbor in this paragraph, can you please tell me if this refers to another minio process on top of the same host node? 

_With in the same MinIO instance locking is handled by existing in-memory namespace locks (**sync.RWMutex** et. al).  To synchronize locks between many MinIO instances we leverage POSIX `fcntl()` locks on Unixes and on Windows `LockFileEx()` Win32 API. Requesting write lock block if there are any read locks held by **neighboring MinIO instance on the same path**. So does the read lock if there are any active write locks in-progress._

**Comment by klauspost** [comment, marked as answer]
This is heavily outdated. It simply refers to other servers serving the same path. But it is no longer relevant.
